src/pages/resultsView/plots/PlotsTab.tsx

```diff
@@ -2138,7 +2161,7 @@ export default class PlotsTab extends React.Component<IPlotsTabProps,{}> {
         invoke: ()=>{
             const horzAxisData = this.horzAxisDataPromise.result;
             const vertAxisData = this.vertAxisDataPromise.result;
-            if (!horzAxisData || !vertAxisData) {
+            if (!horzAxisData || !vertAxisData || (this.showClinicalAttrsWithTreatments && !this.horzGeneOptions.isComplete)) {
```

Comment:
It is correct like it is, but to make it more readable it should state:
```
if (!horzAxisData || !vertAxisData || (this.showClinicalAttrsWithTreatments && !this.utilityMenuGeneOptions.isComplete)) {
```
This means that `this.utilityMenuGeneOptions` should be converted into a `readonly utilitiyMenuGeneOptions = remoteData...` construct.

Reply:
@pvannierop Thanks for your input! I have updated the PR.